packages/editor/src/components/post-publish-panel/maybe-category-panel.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { some } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { PanelBody } from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import HierarchicalTermSelector from '../post-taxonomies/hierarchical-term-selector';
+import { store as editorStore } from '../../store';
+
+function MaybeCategoryPanel() {
+	const hasNoCategory = useSelect( ( select ) => {
+		const postType = select( editorStore ).getCurrentPostType();
+		const categoriesTaxonomy = select( coreStore ).getTaxonomy(
+			'category'
+		);
+		const defaultCategorySlug = 'uncategorized';
+		const defaultCategory = select( coreStore ).getEntityRecords(
+			'taxonomy',
+			'category',
+			{
+				slug: defaultCategorySlug,
+			}
+		)?.[ 0 ];
+
+		const categories =
+			categoriesTaxonomy &&
+			select( editorStore ).getEditedPostAttribute(
+				categoriesTaxonomy.rest_base
+			);
+
+		return (
+			categoriesTaxonomy !== undefined &&
+			defaultCategory &&
+			some( categoriesTaxonomy.types, ( type ) => type === postType ) &&
+			( categories?.length === 0 ||
+				( categories?.length === 1 &&
+					defaultCategory.id === categories[ 0 ] ) )
```

Comment:
This condition is a little hard to parse. Maybe we can split it into multiple returns since we're dealing with booleans?

Reply:
I added a comment and extracted one part of the condition to clarify it.

Reply:
Thanks.